### PR TITLE
Show multiple images

### PR DESCRIPTION
**Pull request by leandroalonso**
Ref #13102

This PR adds the capability to show multiple images and to navigate between them on the Media Editor. Editing a selected image and tapping Done will be addressed in a future PR.

### To test

First, in `GutenbergViewController.swift` replace the function `gutenbergDidRequestFullscreenImage` by this one:

```swift
    func gutenbergDidRequestFullscreenImage(with mediaUrl: URL) {
        gutenbergDidRequestMediaEditor(with: mediaUrl, callback: { _ in })
    }
```

And inside `gutenbergDidRequestMediaEditor` give the `image` multiple times to the MediaEditor, like this (line 447):

```swift
let mediaEditor = WPMediaEditor([image, image, image, image, image, image, image, image, image, image])
```

You can test the same thing in Aztec, just pass the same image multiple times.

#### Swipe between the images

You can swipe between the images, the selected thumbnail (at the bottom) should be updated accordingly.

#### Navigate using the thumbnails

Tapping one thumbnail should display the image at the top. If the image isn't already loaded an activity indicator should appear.

#### Portrait and landscape, iPhone and iPad

If you rotate the device the interface should update accordingly.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/MediaEditor/MediaEditorHub.swift [hunk 1]
```diff
@@ -78,16 +117,128 @@ class MediaEditorHub: UIViewController {
...
+        selectedThumbIndex = indexPath.row
+        imagesCollectionView.scrollToItem(at: indexPath, at: .right, animated: true)
+    }
+
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard scrollView == imagesCollectionView, isUserScrolling else {
+            return
+        }
+
+        let index = Int(round(scrollView.bounds.origin.x / imagesCollectionView.frame.width))
```
Comment (diegoreymendez) [inline WordPress/MediaEditor/MediaEditorHub.swift]: I'm not sure if I'm reading this right, but if I am, you can use this instead https://developer.apple.com/documentation/uikit/uicollectionview/1618020-indexpathsforvisibleitems

```swift
thumbsCollectionView.indexPathsForVisibleItems[0]
```

File: WordPress/MediaEditor/MediaEditorHub.swift [hunk 2]
```diff
@@ -78,16 +117,128 @@ class MediaEditorHub: UIViewController {
...
+
+// MARK: - UICollectionViewDataSource
+
+extension MediaEditorHub: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return numberOfThumbs
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        // Thumbs Collection View
```
Comment (diegoreymendez) [inline WordPress/MediaEditor/MediaEditorHub.swift]: Suggestion: Whenever I see comments like this one and the one below (`// Images Collection View`), I tend to believe code can be separated even further, in order to simplify things and improve readability.

Something like:

```swift
func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {

    if collectionView == thumbsCollectionView {
        return cellForThumbsCollectionView(...)
    } else {
        return cellForImagesCollectionView(...)
    }
}
```

File: WordPress/MediaEditor/MediaEditor.swift [hunk 3]
```diff
@@ -131,29 +164,29 @@ public class MediaEditor: UINavigationController {
         view.layer.add(transition, forKey: nil)
     }
 
-    private func thumbnailAvailable(_ thumb: UIImage?) {
-        guard let thumb = thumb, image == nil else {
```
[the remark this answers is not in the document]
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditor.swift]: It is! This logic only worked for single images, and now we switched for the possibility of many images.

[This line](https://github.com/wordpress-mobile/WordPress-iOS/pull/13171/files#diff-1e86b82ea7f4582213468a01b6c12593R82) in `MediaEditorHub` makes sure that a thumb does not replace a full quality image.

Also, we have a unit test for this: https://github.com/wordpress-mobile/WordPress-iOS/pull/13171/files#diff-c0ddf5f907148e014521b0092d34df4bR184

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Here we're calculating which thumb should be selected based on the scroll in the `imagesCollectionView`.

You can quickly test this by slowing swiping to the next image. When you're seeing more than 50% of the next cell, the corresponding thumb is automatically selected.

But this code is a little bit confusing. I will refactor it to make it more meaningful.

Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Oh, now that I'm reading again I guess your suggestion was to use:

```swift
thumbsCollectionView.indexPathsForVisibleItems[0]
```

However, I tried it and it doesn't work as I intended. To move to the correct thumbnail I'll need to check the direction of the scroll and check the `indexPathsForVisibleItems`.

File: WordPress/MediaEditor/MediaEditor.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply WordPress/MediaEditor/MediaEditor.swift]: Thanks!  That makes sense.

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 1 above)
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Changed in a536e331ebf8d94e1c327451a6e5655405c42a3b.

Let me know what you think.

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 2 above)
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Done in 59c5776e7bd784c6e543e575f2e261dabb388217

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Looks fine to me.

File: WordPress/MediaEditor/MediaEditorHub.swift [hunk 4]
```diff
@@ -78,16 +117,134 @@ class MediaEditorHub: UIViewController {
...
+        if collectionView == thumbsCollectionView {
+            return cellForThumbsCollectionView(cellForItemAt: indexPath)
+        }
+
+        return cellForImagesCollectionView(cellForItemAt: indexPath)
+    }
+
+    private func cellForThumbsCollectionView(cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = thumbsCollectionView.dequeueReusableCell(withReuseIdentifier: Constants.thumbCellIdentifier, for: indexPath) as? MediaEditorThumbCell else {
+            return UICollectionViewCell()
```
Comment (diegoreymendez) [inline WordPress/MediaEditor/MediaEditorHub.swift]: This case happening would likely mean that the identifier provided above is not right, or it doesn't match the one we used for registering the cells.  Either way, this case triggering would signal something's off.

As such I'd suggest to log an error here, before returning.

If you use this method you can log an error with Sentry without that error killing the App or interrupting the user:

https://github.com/Automattic/Automattic-Tracks-iOS/blob/b8e747a37cf5731c77ba7832e3ef54f540740e60/Automattic-Tracks-iOS/CrashLogging.swift#L110

File: WordPress/MediaEditor/MediaEditorHub.swift [hunk 5]
```diff
@@ -78,16 +117,134 @@ class MediaEditorHub: UIViewController {
...
+
+        cell.thumbImageView.image = availableThumbs[indexPath.row]
+        indexPath.row == selectedThumbIndex ? cell.showBorder(color: selectedColor) : cell.hideBorder()
+
+        return cell
+    }
+
+    private func cellForImagesCollectionView(cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = imagesCollectionView.dequeueReusableCell(withReuseIdentifier: Constants.imageCellIdentifier, for: indexPath) as? MediaEditorImageCell else {
+            return UICollectionViewCell()
```
Comment (diegoreymendez) [inline WordPress/MediaEditor/MediaEditorHub.swift]: Same comment as above.

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 4 above)
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: If we return a `UICollectionViewCell()` directly (without using `dequeueReusableCell`) the app will crash anyway (and the crash will be reported to Sentry).

I refactored those methods to make it simpler and close to other examples we have in the app. :)

File: WordPress/MediaEditor/MediaEditorHub.swift (hunk 5 above)
Comment (leandroalonso) [reply WordPress/MediaEditor/MediaEditorHub.swift]: Same reply as above 😄

**Comment by leandroalonso** [comment]
@diegoreymendez since the comments you made also apply to the next PR I'm merging this one. If you have anything else feel free to point on the new one. :)

Hope that's not a problem!

**Pull request merged**